core/shared/src/main/scala/org/virtuslab/yaml/YamlCodec.scala

```diff
@@ -3,6 +3,25 @@ package org.virtuslab.yaml
 /**
  * A type class that provides both-way conversion between [[Node]] and [[T]]
  */
-trait YamlCodec[T] extends YamlDecoder[T] with YamlEncoder[T]
+trait YamlCodec[T] extends YamlDecoder[T] with YamlEncoder[T] { self =>
 
-object YamlCodec extends YamlCodecCompanionCrossCompat
+  def imap[T1](f: T => T1)(g: T1 => T): YamlCodec[T1] =
+    YamlCodec.from(self.map(f), self.contramap(g))
+
+  def iemap[T1](f: T => Either[ConstructError, T1])(g: T1 => T): YamlCodec[T1] =
```

Comment:
is there a better name for this shape of iso map? iemap is actively hostile (imap is too, tbh) if you're not familiar with ct and/or haskell